src/parser/mod.rs

```diff
@@ -3471,7 +3471,7 @@ impl<'a> Parser<'a> {
                     right
                 };
 
-                if !matches!(
+                if !dialect_of!(self is PostgreSqlDialect) && !matches!(
```

Comment:
can we use the dialect trait for this logic (dialect_of is effectively deprecated internally)? something like `if self.dialect.supports_arbitrary_comparison_in_subquery_operator()`